src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -3589,6 +3637,128 @@ NodeIsRangeTblRefReferenceTable(Node *node, List *rangeTableList)
 }
 
 
+/*
+ * FetchEqualityAttrNumsForRTE fetches the attribute numbers from quals
+ * which:
+ * - has equality operator
+ * - belongs to rangeTableEntry with rteIndex
+ */
+List *
+FetchEqualityAttrNumsForRTE(Node *node)
+{
+	if (node == NULL)
+	{
+		return NIL;
+	}
+	if (IsA(node, List))
+	{
+		return FetchEqualityAttrNumsForList((List *) node);
+	}
+	else if (IsA(node, OpExpr))
+	{
+		return FetchEqualityAttrNumsForRTEOpExpr((OpExpr *) node);
+	}
+	else if (IsA(node, BoolExpr))
+	{
+		return FetchEqualityAttrNumsForRTEBoolExpr((BoolExpr *) node);
+	}
+	return NIL;
+}
+
+
+/*
+ * FetchEqualityAttrNumsForList fetches the constant equality numbers
+ * from the given node list.
+ */
+static List *
+FetchEqualityAttrNumsForList(List *nodeList)
+{
+	List *attributeNums = NIL;
+	Node *node = NULL;
+	bool hasAtLeastOneEquality = false;
+	foreach_ptr(node, nodeList)
+	{
+		List *fetchedEqualityAttrNums =
+			FetchEqualityAttrNumsForRTE(node);
+		hasAtLeastOneEquality |= list_length(fetchedEqualityAttrNums) > 0;
+		attributeNums = list_concat(attributeNums, fetchedEqualityAttrNums);
+	}
+
+	/*
+	 * the given list is in the form of AND'ed expressions
+	 * hence if we have one equality then it is enough.
+	 * E.g: dist.a = 5 AND dist.a > 10
+	 */
+	if (hasAtLeastOneEquality)
+	{
+		return attributeNums;
+	}
+	return NIL;
+}
+
+
+/*
+ * FetchEqualityAttrNumsForRTEOpExpr fetches the attribute numbers from opExpr
+ * which:
+ * - has equality operator
+ * - belongs to rangeTableEntry with rteIndex
```

Comment:
where do we do this second check?

Reply:
should be removed, we were doing that when we were using query tree, now we are using base restrict info so all of them already belong to the right variable.